packages/compose/src/hooks/use-merge-refs/index.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import mergeRefs from 'react-merge-refs';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/** @typedef {import('@wordpress/element').RefObject} RefObject */
+
+/**
+ * Merges the given refs into a new ref, which will persist for the full
+ * lifetime of the component, as a ref should be.
+ *
+ * @param  {...RefObject|Function} refs Ref objects or callbacks to merge.
+ *
+ * @return {Function} A new ref callback.
+ */
+export default function useMergeRefs( ...refs ) {
+	return useCallback( mergeRefs( refs ), [] );
```

Comment:
I guess the problem of this implementation is that it won't update the ref if say the "callback" changes. Should we spread `refs` as the dependencies of useCallback?

Reply:
Yes, that makes sense. Will update.

Reply:
Dependencies added in 02041d7.